### PR TITLE
fix mypy for extensions

### PR DESCRIPTION
**Pull request by dni**
- explicitly exclude all extensions from mypy
- fix example extension mypy
- fix subdomains extension mypy + 1 type error fixed
- fix mypy discordbot
- fix mypy copilot

### CODE REVIEW

**Comment by callebtc** [comment]
Why do you exclude the extensions explicitly?

**Comment by dni** [comment]
i exclude them so the new ones willl have to pass mypy, while we slowly keep removing them altogether.

File: lnbits/extensions/copilot/crud.py [hunk 1]
```diff
@@ -71,15 +71,16 @@ async def update_copilot(
...
     items = [f"{field[1]}" for field in data]
-    items.append(copilot_id)
-    await db.execute(f"UPDATE copilot.newer_copilots SET {q} WHERE id = ?", (items))
+    if copilot_id:
+        items.append(copilot_id)
+    await db.execute(f"UPDATE copilot.newer_copilots SET {q} WHERE id = ?", (items,))
     row = await db.fetchone(
         "SELECT * FROM copilot.newer_copilots WHERE id = ?", (copilot_id,)
     )
     return Copilots(**row) if row else None
```
Comment (callebtc) [inline lnbits/extensions/copilot/crud.py:80]: This function doesn't look like the `copilot_id` is actually optional. Won't this just crash if you don't pass it?

Comment (dni) [reply lnbits/extensions/copilot/crud.py]: changed it, ```copillot_id``` is now not ```Optional[str]```

File: lnbits/extensions/discordbot/crud.py [hunk 2]
```diff
@@ -65,8 +65,9 @@ async def get_discordbot_users(user_id: str) -> List[Users]:
 
 async def delete_discordbot_user(user_id: str) -> None:
     wallets = await get_discordbot_wallets(user_id)
-    for wallet in wallets:
-        await delete_wallet(user_id=user_id, wallet_id=wallet.id)
+    if wallets:
+        for wallet in wallets:
+            await delete_wallet(user_id=user_id, wallet_id=wallet.id)
```
Comment (callebtc) [inline lnbits/extensions/discordbot/crud.py:70]: This shouldn't be necessary? The loop only executes if `len(wallets)>0`.

**Pull request merged**